tests/test_connector.py

```diff
@@ -623,11 +631,27 @@ async def _resolve_host(host, port, traces=None):
     os_error = certificate_error = ssl_error = fingerprint_error = False
     connected = False
 
+    async def start_connection(*args, **kwargs):
+        nonlocal os_error, certificate_error, ssl_error, fingerprint_error
+        nonlocal connected
+
+        addr_infos: List[AddrInfoType] = kwargs["addr_infos"]
+
+        first_addr_info = addr_infos[0]
+        first_addr_info_addr = first_addr_info[-1]
+        addrs_tried.append(first_addr_info_addr)
+
+        mock_socket = mock.Mock()
```

Comment:
Similar with mocks like this. If we have a class to base it off, then we can hopefully do `create_autospec(Socket, spec_set=True, instance=True)`

Reply:
349203019c80ce4af4e3514eff3e8976c4e485a5